@msh/msh.m

```diff
@@ -956,6 +956,9 @@ function write(obj,fname,type)
                     disp(['Deleted ' num2str(LT-size(obj.t,1)) ...
                           ' bad boundary elements'])
                 end
+                % kjr, collapse small triangles together 
+                [obj.p,obj.t] = ...
+                    collapse_thin_triangles(obj.p,obj.t,opt.db); 
```

Comment:
Note that when you execute collapse, this can alter the node numbering and thus affect the mapping of the bathymetry data to the cleaned mesh later on. In the examples I've tried, after cleaning outside of the meshgen (i.e., after bathy is interpolated) leads to a mess with the bathymetry

Reply:
Also, if you build a floodplain onto a mesh with a weir, the collapse thin will decimate the weir. You can try by adding a weir to the Houston GBAY example  with floodplain.

Reply:
It seems that if you are meshing with a weir, it's best to use a custom configuration for clean. For some reason lowering the djc to 0.01 from 0.25 kept the weir in tact. When it was kept on, the clean destroyed the weir even though it was being constrained during the mesh generation.

Reply:
The node numbering was affected even without the collapse routine. We've always had a problem with bathymetry getting screwed during cleaning. So we always have just told people (or just ourselves) to only interp bathymetry on after cleaning. Alternatively we need a mapping routine in there.

Reply:
> It seems that if you are meshing with a weir, it's best to use a custom configuration for clean. For some reason lowering the djc to 0.01 from 0.25 kept the weir in tact. When it was kept on, the clean destroyed the weir even though it was being constrained during the mesh generation.

Yes, you can set djc to 0. When cleaning with a weir I would use the passive option which uses djc = 0. This is not an issue, merely you need to select the right option.

Reply:
> Note that when you execute collapse, this can alter the node numbering and thus affect

The node numbering was affected even without the collapse routine. We've always had a problem with bathymetry getting screwed during cleaning. So we always have just told people (or just ourselves) to only interp bathymetry on after cleaning. Alternatively we need a mapping routine in there.

Reply:
Yea, I had forgotten that happens. I'll create an issue.